roles/ansible-ssh/tasks/main.yml

```diff
@@ -0,0 +1,31 @@
+---
+- name: Remove prior ssh config info
+  local_action: file path=/tmp/ansible-ssh-config state=absent
+  changed_when: false
+
+- name: Gather ssh config info for Vagrant VM
+  local_action: shell vagrant ssh-config --host {{ inventory_hostname }} >> /tmp/ansible-ssh-config
+  when: "'development' in group_names"
```

Comment:
This task must be ready to skip for staging/production hosts for whom we don't want to run `vagrant ssh-config`. I had to run this ansible-ssh role run in `server.yml` too because once `ansible.cfg` specifies a ssh config file, the file must be present and only that config file will be used. This doesn't mean you'll be without info from your regular `~/.ssh/config`, however, because this new config file will suck in whatever ssh config "includes" files you specify in `ssh_config_files` (defined in `group_vars/all/main.yml`).